oeps/oep-0006.rst

```diff
@@ -0,0 +1,222 @@
+* https://groups.google.com/d/msg/edx-code/ywjXV0wzQiw/FRzaK5nTAgAJ
+* On 2016-07-08, the author met with several stakeholders from edX (Calen Pennington, David Ormsbee, and Robert Raposa) to discuss course-scoped XBlock fields. The discussion tended toward using "context" as an additional dimension (see "Context as a New Scope Dimension" in "Rejected Alternatives"), but we did not think of any use cases for 8 of the 12 additional scopes that would be created by defining an additional dimension.
+
+Other related proposals of note include:
+
+* https://github.com/edx/XBlock/pull/317 - XBlock field sharing
+
+
+Backward Compatibility
+=======================
+
+This OEP introduces a new field scope but does not remove or modify any existing functionality. XBlock authors that add new fields to their XBlock may opt-in to using this feature.
+
+XBlock runtimes may choose to implement context-scoped fields or not. Attempting to access a context-scoped field in a runtime that does not support such fields should simply return the default value of the field. Attempting to modify such a context-scoped field in a runtime that doesn't support such fields should raise an ``InvalidScopeError``.
+
+Some existing XBlocks currently use workarounds for the lack of context-scoped fields, such as storing data in the fields of the root 'course' XBlock. Converting such existing techniques to use context-scoped fields instead would be best achieved with an XBlock data migration API, which currently does not exist and is outside the scope of this OEP (however, see `here`_ for an example implementation).
+
+.. _here: https://github.com/open-craft/problem-builder/blob/c5879dd/problem_builder/v1/xml_changes.py
+
+Details on the current implementation of field scopes in edx-platform can be found in `this gist`_. The `field test XBlock`_ can be used to test how a runtime behaves with different field scopes.
+
+.. _this GIST: https://gist.github.com/bradenmacdonald/b576c59e46af8df9949fa84b3bc7a4b6
+.. _field test XBlock: https://github.com/open-craft/xblock-field-test
+
+Open Questions
+==============
+
+#. How can an XBlock specify whether a context-scoped field should participate
+   in any sort of auto-generated "course preferences" UI (for
+   BlockScope.CONTEXT_TYPE|UserScope.ONE fields) or "course advanced settings"
+   UI (for BlockScope.CONTEXT_TYPE|UserScope.NONE fields)?
+
+   * This may be outside the scope of this OEP, and would best be addressed at
+     the same time as a standard API for definining editor UI options such as
+     ``allow_reset``, ``multiline_editor``, ``list_style``, and
+     ``values_provider`` as implemented in `xblock-utils`_.
+
+#. The current proposal does not support inheritance (i.e. a "context" in the LMS would always be a course, not a "section" or any smaller unit, and field values don't propagate from parent blocks to children), so it is not a replacement for `InheritanceMixin`_. Should it be?
```

Comment:
I think not participating in inheritance is the right decision. In my mind, context-scoped things are always global in the context.